core/src/components/select/select.tsx

```diff
@@ -765,15 +765,16 @@ export class Select implements ComponentInterface {
 
   private onClick = (ev: UIEvent) => {
     const target = ev.target as HTMLElement;
-    const targetElSlot = target.getAttribute('slot');
+    const closestSlot = target.closest('[slot]');
+    const slotName = closestSlot === null ? "" : closestSlot.getAttribute('slot');
```

Comment:
Optional: You can skip the `slotName` step and just do `target.closest('[slot="start"], [slot="end"]')` to only get the start/end slots. From there, you can check if `closestSlot !== null` instead of checking the slot names below.